pkg/credentialprovider/plugin/config.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth"
+)
+
+// readCRIAuthPluginConfigFile receives a path to a config file and decodes it
+// into the internal CRIAuthPluginConfig type.
+func readCRIAuthPluginConfigFile(configPath string) (*criauth.CRIAuthPluginConfig, error) {
+	if configPath == "" {
+		return nil, fmt.Errorf("CRI auth plugin config path is empty")
+	}
+
+	data, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read external registry credential provider configuration from %q [%v]", configPath, err)
```

Comment:
very minor, just curious why it was made this way. Why is it `[%v]` instead of what typically used for wrapping errors: `: %v`

Reply:
no reason, will update